modules/server/src/main/scala/higherkindness/compendium/http/RootService.scala

```diff
@@ -56,15 +56,21 @@ object RootService {
         (for {
           protocolId <- validateProtocolId(id)(ProtocolIdentifierError)
           protocol   <- CompendiumService[F].recoverProtocol(protocolId)
-          resp       <- protocol.fold(NotFound())(Ok(_))
+          resp       <- protocol.fold(NotFound())(mp => Ok(mp.protocol))
         } yield resp).recoverWith {
           case idError: ProtocolIdentifierError => BadRequest(ErrorResponse(idError.message))
           case _                                => InternalServerError()
         }
 
-      case GET -> Root / "protocol" / _ / "generate" :? TargetQueryParam(_) =>
-        NotImplemented()
-
+      case GET -> Root / "protocol" / id / "generate" :? TargetQueryParam(target) =>
```

Comment:
"generate" sounds like a `POST` to me?

Reply:
Not exactly, for the moment we are converting files that exist in the db. So, when you want to translate from one IDL to another, an original schema must exist in the DB (and Storage too), and that's the schema that will be translated. So you need to post nothing with the request. However, maybe `generate` is a confusing name, if so, I have no problem to change it to another more descriptive 🙂

Reply:
Maybe `translate`?

Reply:
The files that you are converting from the db, are persisted in the db itself?

Reply:
Nope, at least for the moment, the files are in a storage system, which is a folder where the file is saved. As far as I know, this is something that will change, is only a temporal solution